packages/presentation/lib/screen/home/home_bloc.dart

```diff
@@ -1,21 +1,53 @@
+import 'package:domain/usecase/get_movie_anticipated_response_usecase.dart';
 import 'package:presentation/base/bloc.dart';
+import 'package:domain/usecase/get_movie_trending_response_usecase.dart';
+import 'package:presentation/screen/home/home_screen.dart';
+import 'package:presentation/screen/home/movie_model.dart';
+import '../details/details_page.dart';
 import 'home_data.dart';
 
-abstract class HomeBloc extends Bloc {
-  factory HomeBloc() => _HomeBloc();
+abstract class HomeBloc extends Bloc<HomeScreenArguments, HomeData> {
+  factory HomeBloc(
+    GetMovieTrendingResponseUseCase getMovieTrendingResponseUseCase,
+    GetMovieAnticipatedResponseUseCase getMovieAnticipatedResponseUseCase,
+  ) =>
+      _HomeBloc(
+          getMovieTrendingResponseUseCase, getMovieAnticipatedResponseUseCase);
 
   void onItemTapped(int index);
+
+  void openDetailsPage(MoviePresentation movie);
 }
 
-class _HomeBloc extends BlocImpl implements HomeBloc {
+class _HomeBloc extends BlocImpl<HomeScreenArguments, HomeData>
+    implements HomeBloc {
   int? selectedIndex = 1;
-  final _screenData = HomeData.init();
+  HomeData _screenData = HomeData.init();
+  final GetMovieTrendingResponseUseCase _getMovieTrendingResponseUseCase;
+  final GetMovieAnticipatedResponseUseCase _getMovieAnticipatedResponseUseCase;
 
-  _HomeBloc();
+  _HomeBloc(this._getMovieTrendingResponseUseCase,
+      this._getMovieAnticipatedResponseUseCase);
 
   @override
-  void initState() {
+  void initState() async {
     super.initState();
+    _updateData(isLoading: true);
+    final responseAnticipated = await _getMovieAnticipatedResponseUseCase();
+    final responseTrending = await _getMovieTrendingResponseUseCase();
+    final List<MoviePresentation> moviesTrending = responseTrending
+        .map((e) => MoviePresentation.fromMovie(e.movie))
+        .toList();
+    final List<MoviePresentation> moviesAnticipated = responseAnticipated
+        .map((e) => MoviePresentation.fromMovie(e.movie))
+        .toList();
+    _screenData = HomeData(0, moviesTrending, moviesAnticipated);
+    _updateData(isLoading: false);
+  }
+
+  @override
+  void initArgs(HomeScreenArguments arguments) {
+    super.initArgs(arguments);
     _updateData();
   }
```

Comment:
if there's no logic in initArgs, I don't think there's no need in it's overriding. 

initState updates screen with some data and then u update screen with the same data in initArgs. seems to be redundant